### PR TITLE
FLPATH 1048: Extend software template to offer the CI pipeline (Tekton option)

### PR DESCRIPTION
**Pull request by dmartinol**
- Add CI option to select b/w `Tekton `(default) and `None`
- In case of `Tekton`, also generates a `tekton` folder with the workflow templates to generate the triggering resources
- In case of `Tekton`, also generates a GitHub workflow folder to deploy the  resources
- The deploy workflow is invoked by the template

### CODE REVIEW

File: scaffolder-templates/basic-workflow/.github/workflows/deploy.yaml [hunk 1]
```diff
@@ -0,0 +1,44 @@
+# Requirement: add a K8S_SECRET secret to your organization with the SA token
+# oc get secret backstage-k8s-token-4l5xv -n backstage-system -oyaml > secret.yaml
+# gh secret --repo parodos-dev/test-swf set K8S_SECRET <secret.yaml
```
Comment (masayag) [inline scaffolder-templates/basic-workflow/.github/workflows/deploy.yaml:3]: Should we specify here the requirement for the SA token (e.g. which RBAC is required)?

File: scaffolder-templates/basic-workflow/template.yaml [hunk 2]
```diff
@@ -89,20 +94,48 @@ spec:
...
+          enum:
+            - none
+            - tekton
+          enumNames:
+            - None
+            - Tekton
+        clusterDomain:
+          title: Cluster domain
+          type: string
+          default: cluster-jkrpp.dynamic.redhatworkshops.io
```
Comment (masayag) [inline scaffolder-templates/basic-workflow/template.yaml:123]: no need for a default for this field.

File: scaffolder-templates/basic-workflow/template.yaml [hunk 3]
```diff
@@ -89,20 +94,48 @@ spec:
...
+          description: Cluster domain
```
Comment (masayag) [inline scaffolder-templates/basic-workflow/template.yaml:124]: should we specify in this context this cluster domain is for the tekton pipeline so it won't be mixed with being also the cluster to run the workflows?

File: scaffolder-templates/basic-workflow/.github/workflows/deploy.yaml (hunk 1 above)
Comment (dmartinol) [reply scaffolder-templates/basic-workflow/.github/workflows/deploy.yaml:3]: adding a generic `cluster-admin`, but we should review this statement and reduce the permissions to the minimum

Comment (dmartinol) [reply scaffolder-templates/basic-workflow/.github/workflows/deploy.yaml:3]: Added https://github.com/parodos-dev/workflow-software-templates/issues/20 to track the request

File: scaffolder-templates/basic-workflow/template.yaml (hunk 3 above)
Comment (dmartinol) [reply scaffolder-templates/basic-workflow/template.yaml:124]: Ok, even if having workflows and tekton on different clusters is something we have never experiment so far (probably the ArgoCD will need its own cluster configuration, in this multi-cluster deployment)

**Pull request merged**